src/libraries/System.IO.Pipelines/src/PACKAGE.md

```diff
@@ -2,43 +2,113 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+A library designed to make it easier to do high-performance I/O.
 
+Apps that parse streaming data are composed of boilerplate code having many specialized and unusual code flows.
+The boilerplate and special case code is complex and difficult to maintain.
+
+`System.IO.Pipelines` was architected to:
+
+* Have high performance parsing streaming data.
+* Reduce code complexity.
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Single producer/single consumer byte buffer management.
+* Reduction in code complexity and boilerplate code associated with I/O operations.
 
 ## How to Use
 
 <!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
 
+The `Pipe` class can be used to create a `PipeWriter`/`PipeReader` pair.
+All data written into the `PipeWriter` is available in the `PipeReader`.
+
+```csharp
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Text;
+
+// This could be an external source like a Socket, for this example it's just a string
+byte[] input = Encoding.UTF8.GetBytes("Hello, Pipelines!");
+
+var pipe = new Pipe();
+Task writing = FillPipeAsync(pipe.Writer, input);
+Task reading = ReadPipeAsync(pipe.Reader);
+
+await Task.WhenAll(reading, writing);
+
+static async Task FillPipeAsync(PipeWriter writer, byte[] input)
+{
+    for (int i = 0; i < input.Length; i++)
+    {
+        Memory<byte> memory = writer.GetMemory(1);
+        memory.Span[0] = input[i];
+        writer.Advance(1);
+
+        // Make the data available to the PipeReader.
+        await writer.FlushAsync();
+    }
```

Comment:
This example looks really inefficient.

Reply:
I tried to create a simple implementation of the workings of `Pipe` but have no further experience with it in practice.

The example that is part of the _conceptual documentation_ link (<https://learn.microsoft.com/en-us/dotnet/standard/io/pipelines#pipe-basic-usage>) requires quite a large explanation, which didn't seem appropriate for the readme of the package itself.

I'm open for suggestions.

Reply:
cc @area-system-io

Reply:
cc @dotnet/area-system-io